libs/ui/src/_utils/typeguards.ts

```diff
@@ -0,0 +1,3 @@
+import {WithTypename} from '@leav/utils';
+
+export const hasTypename = <T>(value: any): value is WithTypename<T> => !!value.__typename;
```

Comment:
suggestion: pour un typeguard on pourrait tester juste l'existence de la clé avec` '__typename' in value`
avec !! ça dira qu'il n'est pas là si il est falsy
mais c'est peut être ce que tu voulais à l'usage

Reply:
tu as raison, c'est plus propre comme ça